control/metrics.go

```diff
@@ -250,6 +262,62 @@ func (m *metricType) Unit() string {
 	return m.unit
 }
 
+type catalogedPlugin struct {
```

Comment:
I don't see any. Is there a problem with definition of private structure which implements this interface? It is quite common in other packages as well.